android/src/AllFilesDocumentsProvider.java

```diff
@@ -0,0 +1,338 @@
+        row.add(Document.COLUMN_FLAGS, flags);
+    }
+
+    @Override
+    public ParcelFileDescriptor openDocument(String documentId, String mode, CancellationSignal signal) throws FileNotFoundException {
+        return ParcelFileDescriptor.open(getFileForDocId(documentId), ParcelFileDescriptor.parseMode(mode));
+    }
+
+    @Override
+    public void deleteDocument(final String documentId) throws FileNotFoundException {
+        File file = getFileForDocId(documentId);
+        if (file.delete()) {
+            File parentFile = file.getParentFile();
+            if (parentFile != null) {
+                String parentDocumentId = getDocIdForFile(file.getParentFile());
+                notifyChildDocumentsChange(parentDocumentId);
+            }
+        } else {
+            throw new FileNotFoundException("Failed to delete document id : " + documentId);
+        }
+    }
+
+    @Override
+    public Cursor querySearchDocuments(String rootId, String query, String[] projection) throws FileNotFoundException {
+        final MatrixCursor result = new MatrixCursor(resolveDocumentProjection(projection));
+        final File parent = getContext().getExternalFilesDir(null);
+        final LinkedList<File> pending = new LinkedList<>();
+        final String lowerQuery = query.toLowerCase();
+
+        // Start by adding the parent to the list of files to be processed
+        pending.add(parent);
+
+        // Do while we still have unexamined files, and fewer than the max search results
+        while (!pending.isEmpty() && result.getCount() < MAX_SEARCH_RESULTS) {
+            // Take a file from the list of unprocessed files
+            final File file = pending.removeFirst();
+            if (file.isDirectory()) {
+                // If it's a directory, add all its children to the unprocessed list
+                Collections.addAll(pending, file.listFiles());
+            } else {
+                // If it's a file and it matches, add it to the result cursor.
+                if (file.getName().toLowerCase().contains(lowerQuery)) {
+                    addDocumentRow(result, file);
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String createDocument(String parentDocumentId, String mimeType, String displayName) throws FileNotFoundException {
+        final File parent = getFileForDocId(parentDocumentId);
+        if (!parent.isDirectory()) {
+            throw new IllegalArgumentException("Parent document isn't a directory");
+        }
+        File file = new File(parent, displayName);
+
+        if (Document.MIME_TYPE_DIR.equals(mimeType)) {
+            if (!file.mkdirs()) {
+                throw new FileNotFoundException("Failed to create directory with name " +
+                        displayName + " and documentId " + parentDocumentId);
+            }
+        } else {
+            try {
+                file.createNewFile();
+                file.setWritable(true);
```

Comment:
Should this be:
public boolean setWritable(boolean writable, boolean ownerOnly) 
so therefore true,true ? 
https://www.demo2s.com/android/android-file-setwritable-boolean-writable-boolean-owneronly.html

Reply:
it's : https://developer.android.com/reference/java/io/File#setWritable(boolean)